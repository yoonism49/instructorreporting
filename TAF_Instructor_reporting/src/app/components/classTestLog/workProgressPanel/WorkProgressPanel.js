// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';

const WorkProgressPanel = ({
  title,
  children
}) => (
  <section className="panel">
    <header className="panel-heading">
      {title}
    </header>
    <div className="panel-body table-responsive">
      {children}
    </div>
  </section>
);

WorkProgressPanel.propTypes = {
  title:    PropTypes.string,
  children: PropTypes.node.isRequired
};

WorkProgressPanel.defaultProps = {
  title: 'Work Progress'
};

export default WorkProgressPanel;
