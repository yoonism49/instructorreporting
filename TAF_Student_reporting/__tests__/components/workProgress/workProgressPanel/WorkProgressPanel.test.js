'use strict';

import React                from 'react';
import WorkProgressPanel    from '../../../../src/app/components/workProgress/workProgressPanel/WorkProgressPanel';
import renderer             from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('WorkProgressPanel component', () => {
  const mockProps = {
    title: 'a title for test'
  };

  it('renders as expected', () => {
    const component = renderer.create(
      <div>
        <WorkProgressPanel {...mockProps}>
          <p>
            a work workProgress child
          </p>
        </WorkProgressPanel>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
