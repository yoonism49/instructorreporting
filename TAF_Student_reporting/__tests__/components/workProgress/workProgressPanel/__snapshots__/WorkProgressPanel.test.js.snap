// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkProgressPanel component renders as expected 1`] = `
<div>
  <section
    className="panel"
  >
    <header
      className="panel-heading"
    >
      a title for test
    </header>
    <div
      className="panel-body table-responsive"
    >
      <p>
        a work workProgress child
      </p>
    </div>
  </section>
</div>
`;
